### PR TITLE
Enable rustfmt to format a list of files

### LINKED ISSUE #580: Process a list of files instead of just one
We currently only process a single file, it would be nice to process more. Even better would be to support globs and other wildcards.

See some discussion here https://github.com/rust-lang-nursery/rustfmt/pull/570#issuecomment-155001242 and below

### PR DESCRIPTION
**Pull request by JanLikar**
Fix #580 by allowing rustfmt to accept a list of files. This also enables usage of shell wildcard expansion, although notably this does not work with cmd.exe on Windows. For example: `rustfmt *.rs` will format all rust files in the current working directory.

  - Change usage text to show rustfmt will accept a list of files
  - Change `Using rustfmt config file: {}` message to `Using rustfmt config file {} for {}`
  - Change `Operation::Format(PathBuf, WriteMode)` to `Operation::Format(Vec<PathBuf>, WriteMode)`
  - Loop through `Vec<PathBuf>`, load config and call 'run' for each path

### CODE REVIEW

File: src/bin/rustfmt.rs [hunk 1]
```diff
@@ -29,8 +29,8 @@ use getopts::Options;
 
 /// Rustfmt operations.
 enum Operation {
-    /// Format a file and its child modules.
-    Format(PathBuf, WriteMode),
+    /// Format files and its child modules.
```
Comment (nrc) [inline src/bin/rustfmt.rs]: nit: and their child modules

File: src/bin/rustfmt.rs [hunk 2]
```diff
@@ -144,7 +148,7 @@ fn main() {
 }
 
 fn print_usage(opts: &Options, reason: &str) {
-    let reason = format!("{}\nusage: {} [options] <file>",
+    let reason = format!("{}\nusage: {} [options] <file>...",
```
Comment (nrc) [inline src/bin/rustfmt.rs]: nit: `<file>+` would be clearer to me

Comment (nrc) [inline src/bin/rustfmt.rs]: or maybe `<files>`

File: src/bin/rustfmt.rs [hunk 3]
```diff
@@ -189,5 +193,11 @@ fn determine_operation<I>(opts: &Options, args: I) -> Operation
         None => WriteMode::Replace,
     };
 
-    Operation::Format(PathBuf::from(&matches.free[0]), write_mode)
+    let mut files = Vec::with_capacity(matches.free.len());
+
+    for arg in matches.free {
+        files.push(PathBuf::from(arg));
+    }
```
Comment (nrc) [inline src/bin/rustfmt.rs]: This could be written more succinctly:

```
let files: Vec<_> = matches.free.iter().map(|a| PathBuf::from(a)).collect();
```

**Comment by nrc** [comment]
Thanks for the PR! There are just a few nits to address, otherwise looks good.

File: src/bin/rustfmt.rs (hunk 2 above)
Comment (JanLikar) [inline src/bin/rustfmt.rs]: I don't really agree. GNUutils and others mostly use "<arg>..." or "[arg]...".

Comment (nrc) [inline src/bin/rustfmt.rs]: OK, if there is a convention, I'm happy to stick to it

Comment (JanLikar) [inline src/bin/rustfmt.rs]: Great, I will fix other issued ASAP.

**Comment by nrc** [comment]
Thanks for making the changes!

**Pull request merged**